api/src/main/java/javax/application/event/EventSubscriber.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package javax.application.event;
+
+/**
+ * Base contract for classes that will provide event handlers using their own
+ * event bus.
+ */
+public interface EventSubscriber {
+    /**
+     * Adds an event handler.<p>
+     *
+     * @param handler an event handler. Must not be {@code null}.
+     */
+    void subscribe(Object handler);
```

Comment:
Can this parameter type be EventHandler type? Same comment for the unsubscribe method as well.

Reply:
No, it is not possible. EventHandler is annotation.  You use it to annotate the methods in handler object that will be executed when a event of specific type is triggered.

Reply:
Sorry, you are right, I got confused.